apps/controllers/midibox_ng_v1/src/mbng_event.c

```diff
@@ -4089,6 +4222,33 @@ s32 MBNG_EVENT_ItemReceive(mbng_event_item_t *item, u16 value, u8 from_midi, u8
 	}
 	break;
 
+	  //Mackie C4 Control special case
+	  //considers different encoder rotation accelerations
+      case MBNG_EVENT_ENC_MODE_C4ENC:
+    	  if( item->value != 0x00 ) {
+    		  if(item->value < 0x40){
+    			  switch(item->value){
+    			  case 0x01: incrementer = 1; break;
+    			  case 0x04: incrementer = 2; break;
+    			  case 0x08: incrementer = 4; break;
+    			  case 0x0c: incrementer = 8; break;
+    			  case 0x0f: incrementer = 12; break;
+    			  default:   incrementer = 1; break;
+    			  }
+    		  }
+    		  else{
+    			  switch(item->value){
+    			  case 0x41: incrementer = -1; break;

```

Comment:
same here: range checks would be better

Reply:
What about mbng_enc.c line 223 onwards?

`case MBNG_EVENT_ENC_MODE_C4ENC:
      if(event_incrementer > 0){
    	  switch(event_incrementer){
    	  case 1: item.value = 0x01; break;
    	  case 2: item.value = 0x04; break;
    	  case 4: item.value = 0x08; break;`
And so on..

Is a range check more efficient in that case too?